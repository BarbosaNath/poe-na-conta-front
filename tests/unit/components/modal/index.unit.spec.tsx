import React from 'react';
import Modal from '@components/modal';
import { render } from '@testing-library/react';

describe('Modal', () => {
  const mockedOnClose = vi.fn();
  const renderModal = (isOpen: boolean) =>
    render(
      <Modal isOpen={isOpen} onClose={mockedOnClose}>
        modal
      </Modal>,
    );

  it('should render modal disabled', () => {
    const { container } = renderModal(false);

    expect(container).toMatchSnapshot();
  });

  it('should render modal enabled', () => {
    const { container } = renderModal(true);

    expect(container).toMatchSnapshot();
  });
});
