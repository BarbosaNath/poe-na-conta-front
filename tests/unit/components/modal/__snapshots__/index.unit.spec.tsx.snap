// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal > should render modal disabled 1`] = `<div />`;

exports[`Modal > should render modal enabled 1`] = `
<div>
  <div
    class="lds--modal__wrapper"
  >
    <div
      class="lds--modal__backdrop"
    />
    <div
      class="lds--modal__content"
    >
      modal
    </div>
  </div>
</div>
`;
